packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,244 @@ icon: flame
+    this._id = `mock:${'0'.repeat(64)}:${Math.floor(Math.random() * 10000)}`
+    this._rev = `mock:${'0'.repeat(64)}:${Math.floor(Math.random() * 10000)}`
+    this._root = `mock:${'0'.repeat(64)}:${Math.floor(Math.random() * 10000)}`
     this._amount = amount
   }
 
   transfer(to: string) {
     this._owners = [to]
   }
+
+  setAmount(a: number) {
+    this._amount = a
+  }
 }
+
+const mock = new PaymentMock(7860)
 ```
 
-### Usage
+Now Seller is ready to create and sign the sale transaction using `computer.encode` as shown below. There is a lot going on, so we will break down the arguments below.
 
-Todo: add description here.
+```ts
+const { tx } = await seller.encode({
+  exp: `${Sale} Sale.exec(n, p)`,
+  env: { n: nft._rev, p: mock._rev },
+  mocks: { payment: mock },
+  sighashType: SIGHASH_SINGLE | SIGHASH_ANYONECANPAY,
+  inputIndex: 0,
+  fund: false,
+})
+```
+
+The expression contains the source code of the `Sale` class and an expression that calls the static `exec` function. Note that every time a sale transaction is created, the source code of the `Sale` class is written into the blockchain again. A more efficient approach is to deploy the `Sale` class as a module first. This is described [here](#reducing-fees).
 
 ```ts
-// Create and fund wallet
-const seller = new Computer(RLTC)
-const buyer = new Computer(RLTC)
+  exp: `${Sale} Sale.exec(nft, payment)`,
+```
+
+The next two lines contain the instructions for the mocking system. Seller can use the `mock` object that was created earlier as shown in the code below. The revision of the mock is passed in as the payment revision to the environment; and an object is passed to the `mocks` key that maps the name `payment` to the `mock` object. If an object is mocked up, the Bitcoin Computer will build a transaction without checking that mocked up objects exist on the blockchain and will assume that the value is as in the object passed into `mocks`. Later buyer will update the input that spends the revision of the payment mock with the revision and output number of an actual payment.
+
+```ts
+  env: { nft: nft._rev, payment: mock._rev },
+  mocks: { payment: mock },
+```
+
+To enable Buyer to modify Seller's transaction later, Seller signs the first input with the [sighash type](https://developer.bitcoin.org/devguide/transactions.html#signature-hash-types) `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY`. This means that Seller's signature remains valid even when arbitrary inputs and outputs are added to the transaction. In particular, another user could change or remove the Bitcoin Computer specific meta data that is located in output 3.
+
+Essentially the Seller is saying: you can spend the output containing the nft as long as you keep the locking script and the amount of the first output the same. Seller can confidentially publish such a partially signed transaction as he wants to sell the nft spent by the fist inputs for the amount indicated in the first output and both are signed.
```

Comment:
Slight change and typo correction: Corrected:

"Essentially, the Seller is stating: you can spend the output containing the NFT as long as you maintain the locking script and the amount of the first output unchanged. The Seller can confidently publish such a partially signed transaction as it is intended to sell the NFT spent by the first input for the amount indicated in the first output, and both are signed."